pinot-common/src/main/java/org/apache/pinot/common/utils/DataSchema.java

```diff
@@ -215,7 +236,34 @@ public int hashCode() {
   }
 
   public enum ColumnDataType {
-    INT, LONG, FLOAT, DOUBLE, STRING, BYTES, OBJECT, INT_ARRAY, LONG_ARRAY, FLOAT_ARRAY, DOUBLE_ARRAY, STRING_ARRAY;
+    INT,
+    LONG,
+    FLOAT,
+    DOUBLE,
+    BOOLEAN /* Stored as INT */,
```

Comment:
Should `BOOLEAN` should be stored as `BYTE` to save space?

Reply:
Internally we don't have `BYTE` type, so here we use the smallest type to store `BOOLEAN`.